python/src/HSICEstimatorImplementation_doc.i.in

```diff
@@ -0,0 +1,284 @@
+%define OT_HSICEstimator_doc
+"Base class of HSICEstimator.
+
+Available constructors:
+    HSICEstimator()
+
+    HSICEstimator(covarianceList, X, Y, estimatorType)
+
+    HSICEstimator(covarianceList, X, Y, estimatorType, weightFunction)
+
+Notes
+-----
+This is a base class for the implementation of an HSIC estimator.
+Using *HSICEstimator* is equivalent to using its derived class
+:class:`~openturns.HSICEstimatorConditionalSensitivity`.
+Alternative HSIC-based analyses can be performed by using the other derived classes: :class:`~openturns.HSICEstimatorGlobalSensitivity` and :class:`~openturns.HSICEstimatorTargetSensitivity`.
+
+Parameters
+----------
+covarianceList : list of :class:`~openturns.CovarianceModel`
+    List of all covariance kernels. The :math:`d` first kernels are linked to the input 
+    and the last one is for the output.
+
+X : :class:`~openturns.Sample`
+    The input sample used for the HSIC analysis of dimension :math:`d`.
+
+Y : :class:`~openturns.Sample`
```

Comment:
on input Samples can be a generic "2-d sequence of float"

Reply:
OK